### PR TITLE
Edit campaing application task(Admin and Organaizer)

### PR DESCRIPTION
**Pull request by Martbul**

### CODE REVIEW

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 1]
```diff
@@ -88,8 +94,99 @@ export class CampaignApplicationService {
...
-    return `This action updates a #${id} campaignApplication`
+  async updateCampaignApplication(
+    id: string,
+    personId: string,
+    updateCampaignApplicationDto: UpdateCampaignApplicationDto,
+    isAdminFlag: boolean,
+    organaizerId?: string,
+    files?: Express.Multer.File[],
+  ) {
+    console.log(id)
```
Comment (slavcho) [inline apps/api/src/campaign-application/campaign-application.service.ts:105]: Remove please

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 2]
```diff
@@ -88,8 +94,99 @@ export class CampaignApplicationService {
...
   }
 
-  update(id: string, updateCampaignApplicationDto: UpdateCampaignApplicationDto) {
-    return `This action updates a #${id} campaignApplication`
+  async updateCampaignApplication(
+    id: string,
+    personId: string,
+    updateCampaignApplicationDto: UpdateCampaignApplicationDto,
+    isAdminFlag: boolean,
+    organaizerId?: string,
```
Comment (slavcho) [inline apps/api/src/campaign-application/campaign-application.service.ts:102]: typo: organizerId

File: apps/api/src/campaign-application/campaign-application.service.ts [hunk 3]
```diff
@@ -88,8 +94,99 @@ export class CampaignApplicationService {
...
+
+    if (files) {
+      const existingCampaignApplicationFiles = await this.prisma.campaignApplicationFile.findMany({
+        where: { campaignApplicationId: id },
+      })
+
+      for (const file of existingCampaignApplicationFiles) {
+        await this.s3.deleteObject(this.bucketName, file.id)
+      }
+
```
Comment (slavcho) [inline apps/api/src/campaign-application/campaign-application.service.ts:177]: Why do we delete the previous files?
I'd say the default is that we incrementally add new files.
If the user wants to delete a file - they should do so explicitly one by one.

File: apps/api/src/person/__mock__/personMock.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { Person } from '@prisma/client'
 
 export const personMock: Person = {
-  id: 'e43348aa-be33-4c12-80bf-2adfbf8736cd',
+  id: 'personId',
```
Comment (slavcho) [inline apps/api/src/person/__mock__/personMock.ts:4]: why?

**Review by slavcho — CHANGES_REQUESTED** [review]

File: apps/api/src/campaign-application/campaign-application.controller.ts [hunk 5]
```diff
@@ -67,15 +67,46 @@ export class CampaignApplicationController {
...
+    @UploadedFiles() files: Express.Multer.File[],
     @Param('id') id: string,
     @Body() updateCampaignApplicationDto: UpdateCampaignApplicationDto,
     @AuthenticatedUser() user: KeycloakTokenParsed,
   ) {
-    return this.campaignApplicationService.update(id, updateCampaignApplicationDto)
+    const person = await this.personService.findOneByKeycloakId(user.sub)
+    if (!person) throw new NotFoundException('User is not found')
+
+    let isAdminFlag
```
[the remark this answers is not in the document]
Comment (Martbul) [reply apps/api/src/campaign-application/campaign-application.controller.ts:87]: using person.id

**Review by slavcho — APPROVED** [review]

**Pull request merged**